client/src/Pages/TripDetails.tsx

```diff
@@ -21,15 +30,18 @@ type TripProps = {
 export default function TripDetails({ onClick }: TripProps) {
   const [activeIcon, setActiveIcon] = useState('');
   const [err, setError] = useState<Error>();
-  const [activeEventId, setActiveEventId] = useState<number | undefined>(1);
+  const [activeEventId, setActiveEventId] = useState<number | undefined>(
```

Comment:
You can just write this as:
`const [activeEventId, setActiveEventId] = useState<number>();`
and it will be exactly the same. This is because the default value for `useState` is `undefined` and TS will type the variable as `number | undefined` for you when it sees an empty parameter list in `useState`. This magic happens because of how the types for `useState` are specified.